src/tir/schedule/primitive/fuse_split.cc

```diff
@@ -0,0 +1,483 @@
+      ObjectPtr<BlockRealizeNode> n = CopyOnWrite(realize);
+      n->predicate = n->predicate && predicate_;
+      return BlockRealize(n);
+    } else {
+      return GetRef<BlockRealize>(realize);
+    }
+  }
+
+  /*! \brief The predicate to be added */
+  const PrimExpr& predicate_;
+  /*! \brief whether to add predicate */
+  bool add_predicate_;
+};
+/*! \brief Substitute vars and collect the reuse mapping of opaque blocks */
+class IRSubstituteAndCollectOpaqueBlock : public StmtExprMutator {
+ public:
+  explicit IRSubstituteAndCollectOpaqueBlock(std::function<Optional<PrimExpr>(const Var&)> vmap,
+                                             Map<Block, Block>* opaque_blocks)
+      : vmap_(vmap), opaque_blocks_(opaque_blocks) {}
+
+ private:
+  PrimExpr VisitExpr_(const VarNode* op) final {
+    Var var = GetRef<Var>(op);
+    Optional<PrimExpr> ret = vmap_(var);
+    if (ret.defined()) {
+      return ret.value();
+    } else {
+      return std::move(var);
+    }
+  }
+
+  Stmt VisitStmt_(const BlockRealizeNode* op) final {
+    Stmt res = StmtMutator::VisitStmt_(op);
+    if (op->block->iter_vars.empty()) {
+      const BlockRealizeNode* realize = res.as<BlockRealizeNode>();
+      opaque_blocks_->Set(op->block, realize->block);
+    }
+    return res;
+  }
+
+  /*! \brief The substitute function */
+  std::function<Optional<PrimExpr>(const Var&)> vmap_;
+  /*! \brief The reuse mapping */
+  Map<Block, Block>* opaque_blocks_;
+};
+
+Stmt SubstituteAndCollectOpaqueBlock(Stmt stmt, Map<Block, Block>* opaque_blocks,
+                                     std::function<Optional<PrimExpr>(const Var&)> vmap) {
+  return IRSubstituteAndCollectOpaqueBlock(vmap, opaque_blocks)(std::move(stmt));
+}
+
+/*! \brief Simplify the binding of block realize and update the opaque block reuse mapping*/
+class BlockRealizeRewriter : public StmtExprMutator {
+ public:
+  explicit BlockRealizeRewriter(
+      const std::unordered_map<Var, Range, ObjectPtrHash, ObjectPtrEqual>& loop_map,
+      Map<Block, Block>* opaque_blocks)
+      : opaque_blocks_(opaque_blocks) {
+    loop_map_.insert(loop_map.begin(), loop_map.end());
+  }
+
+ private:
+  Stmt VisitStmt_(const ForNode* op) final {
+    loop_map_[op->loop_var] = Range::FromMinExtent(op->min, op->extent);
+    Stmt res = StmtMutator::VisitStmt_(op);
+    loop_map_.erase(op->loop_var);
+    return res;
+  }
+
+  Stmt VisitStmt_(const BlockRealizeNode* op) final {
+    // skip opaque block and update mapping
+    if (op->iter_values.empty()) {
+      Stmt res = StmtMutator::VisitStmt_(op);
+      const BlockRealizeNode* realize = res.as<BlockRealizeNode>();
+      for (const std::pair<Block, Block>& entry : *opaque_blocks_) {
+        if (entry.second.same_as(op->block)) {
+          opaque_blocks_->Set(entry.first, realize->block);
+          break;
+        }
+      }
+      return res;
+    }
+    auto v = arith::IterMapSimplify(op->iter_values, loop_map_, op->predicate, false);
```

Comment:
```suggestion
    Array<PrimExpr> v = arith::IterMapSimplify(/*indices=*/op->iter_values, /*input_iters=*/loop_map_, /*input_pred=*/op->predicate, /*require_bijective=*/false);
```

Reply:
Perhaps the following looks prettier.
```suggestion
    auto v = arith::IterMapSimplify(/*indices=*/op->iter_values,
                                    /*input_iters=*/loop_map_,
                                    /*input_pred=*/op->predicate,
                                    /*require_bijective=*/false);
```